sycl/doc/syclgraph/SYCLGraphUsageGuide.md

```diff
@@ -680,3 +764,77 @@ execMainGraph.update(updateGraph);
 // ptrA
 myQueue.ext_oneapi_graph(execMainGraph);
 ```
+
+### Graph-Owned Memory Allocations
+
+#### Explicit Graph Example
+
+Using default memory pool.
+
+```c++
+using namespace sycl;
+namespace sycl_ext = sycl::ext::oneapi::experimental;
+
+void* Ptr = nullptr;
+size_t AllocSize = 1024;
+// Add an async_malloc node and capturing the returned pointer in Ptr
+auto AllocNode = Graph.add([&](handler& CGH){
+  Ptr = sycl_ext::async_malloc(CGH, usm::alloc::device, AllocSize);
+});
+
+// Use Ptr in another graph node which depends on AllocNode
+auto OtherNodeA = Graph.add(..., {property::graph::depends_on{AllocNode}});
+// Use Ptr in another node which has an indirect dependency on AllocNode
+auto OtherNodeB = Graph.add(..., {property::graph::depends_on{OtherNodeA}});
+
+// Free Ptr, indicating it is no longer in use at this point in the graph,
+// with a dependency on any leaf nodes using Ptr
+Graph.add([&](handler& CGH){
+  sycl_ext::async_free(CGH, Ptr);
+}, {property::graph::depends_on{OtherNodeB}});
+```
+
+#### Queue Recording Example
+
+Using user-provided memory pool.
+
+```c++
+using namespace sycl;
+namespace sycl_ext = sycl::ext::oneapi::experimental;
+
+void* Ptr = nullptr;
+size_t AllocSize = 1024;
+queue Queue {syclContext, syclDevice};
+
+// Device memory pool with zero init property
+sycl_ext::memory_pool MemPool{syclContext, syclDevice, usm::alloc::device,
+                             {sycl_ext::property::memory_pool::zero_init{}}};
```

Comment:
can we first allocate a buffer, and pass the pointer to MemPool since this is the typical usage that we'll use.

Reply:
Right now in the spec it is not supported (we will respect some properties of the pool as listed in the spec but will not make allocations from the user-provided memory/pool) and it is not within the current scope of graphs support for allocations. This feature is also not implemented for asynchronous allocations outside of graphs (currently device only pools), which makes supporting it difficult for us. 

There are a couple of ways we could potentially approach it but they are not without tradeoffs, and probably require some closer collaboration/examination of your use-case.